addons/website_version/controllers/main.py

```diff
@@ -0,0 +1,155 @@
+# -*- coding: utf-8 -*-
+import datetime
+from openerp import http
+from openerp.http import request
+from openerp.addons.website.controllers.main import Website
+
+
+class Versioning_Controller(Website):
+
+    @http.route(['/website_version/change_version'], type='json', auth="user", website=True)
+    def change_version(self, version_id):
+        request.session['version_id'] = version_id
+        return version_id
+
+    @http.route(['/website_version/create_version'], type='json', auth="user", website=True)
+    def create_version(self, name, version_id):
+        if not name:
+            name = datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S')
+        new_version = request.env['website_version.version'].create({'name': name, 'website_id': request.website.id})
+        if version_id:
+            request.env['website_version.version'].browse(version_id).copy_version(new_version.id)
+        request.session['version_id'] = new_version.id
+        return new_version.id
+
+    @http.route(['/website_version/delete_version'], type='json', auth="user", website=True)
+    def delete_version(self, version_id):
+        version = request.env['website_version.version'].browse(version_id)
+        name = version.name
+        version.unlink()
+        current_id = request.context.get('version_id')
+        if version_id == current_id:
+            request.session['version_id'] = 0
+        return name
+
+    @http.route(['/website_version/check_version'], type='json', auth="user", website=True)
+    def check_version(self, version_id):
+        #To check if the version is in a running or paused experiment
+        Exp = request.env['website_version.experiment']
+        return bool(Exp.search(['|', ('state', '=', 'running'), ('state', '=', 'paused'), ('experiment_version_ids.version_id', '=', version_id)], limit=1))
+    
+    @http.route(['/website_version/all_versions'], type='json', auth="public", website=True)
+    def all_versions(self, view_id):
+        #To get all versions in the menu
+        v = request.env['ir.ui.view'].browse(view_id)
```

Comment:
For this one and every other "one-letter-variable", try to use better naming. I know this is not "a bug", but try to do it. This helps understanding the code, maintaining it. Don't forget other people will probably have to go into your code and update it / fix it / improve it / hack it ;) . So, call this one view, just 3 characters more. Obviously this is valid for your whole PR, I won't spot every one-letter-variable.